packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/TypeProvider.cs

```diff
@@ -111,6 +114,8 @@ private TypeSignatureModifiers GetDeclarationModifiersInternal()
 
         protected virtual TypeProvider[] BuildNestedTypes() => Array.Empty<TypeProvider>();
 
+        protected virtual TypeProvider[] BuildSerializationProviders() => Array.Empty<TypeProvider>();
```

Comment:
I think not every type we want to generate could have a concept of "serialization". So why do we move it here?

Reply:
Every type can have a serialization which is why we moved this to the base class